docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+# Blocks in Widget Areas RFC
+
+This RFC outlines the technical approach that we will take to upgrade the widget-editing areas in `wp-admin/widgets.php` and the Customizer to support blocks.
+
+Adding blocks to widget-editing areas was first announced at WCUS 2018, is a [Gutenberg Phase 2][phase-2] project, and appears in Matt's list of [9 Projects for 2019][2019-projects].
+
+![widgets](https://user-images.githubusercontent.com/1202812/51921660-d4ef0e00-23b5-11e9-86c9-5cb25c224d0b.gif)
+
+[phase-2]: https://github.com/WordPress/gutenberg/issues/13113
+[2019-projects]: https://make.wordpress.org/core/2018/12/08/9-priorities-for-2019/
+
+## Requirements
+
+- Editing blocks in `wp-admin/widgets.php` and the `wp-admin/customize.php` should use the same block editor that `wp-admin/post-new.php` uses.
+- The block editor should read and update blocks in widget-editing areas via the REST API.
+- Upgrading WordPress must not affect the appearance of the user's site, or any of their existing widgets.
+- Existing Core and third-party widgets must remain functional in the new block-based interface.
+- Backwards compatibility must be maintained. That is, themes and plugins that use public widget APIs must remain functional.
```

Comment:
I think we should add another requirement.

It should be possible to "disable" the block-based widgets UI and fallback to the old widgets screen during the transition period.

Reply:
Agreed. Added in b97139aa7.